example_client/camera_client.py

```diff
@@ -0,0 +1,182 @@
+    source = args.video_source
+
+cap = cv2.VideoCapture(source)
+force_exit = False
+
+
+def draw_boxes(frame, result):
+    output = make_ndarray(result.outputs['boxes'])
+    for i in range(0, 100):  # there is returned 200 detections for each image in the batch
+        detection = output[i,:]
+        if detection[4] > 0.3:
+            x_min = int(detection[0])
+            y_min = int(detection[1])
+            x_max = int(detection[2])
+            y_max = int(detection[3])
+            frame = cv2.rectangle(frame,(x_min,y_min),(x_max,y_max),(0,0,255),1)
+    return frame
+
+
+class RequestingThread(threading.Thread):
+    def __init__(self, index):
+        print(f"Initializing requesting thread index: {index}")
+        super().__init__()
+        self.index = index
+        self.input_frame = None
+        self.output_frame = None
+        self.predict_durations = []
+        self.input_ready_event = threading.Event()
+        self.output_ready_event = threading.Event()
+
+    def is_initialized(self):
+        return not (self.input_frame is None and self.output_frame is None)
+
+    def wait_for_input(self):
+        self.input_ready_event.wait()
+        self.input_ready_event.clear()
+
+    def wait_for_result(self):
+        self.output_ready_event.wait()
+        self.output_ready_event.clear()
+
+    def notify_input_ready(self):
+        self.input_ready_event.set()
+
+    def notify_output_ready(self):
+        self.output_ready_event.set()
+
+    def set_input(self, frame):
+        self.input_frame = frame
+        self.notify_input_ready()
+
+    def get_output(self):
+        return self.output_frame
+
+    def get_average_latency(self):
+        return np.average(np.array(self.predict_durations))
+
+    def run(self):
+        print(f"Launching requesting thread index: {self.index}")
+        global force_exit
+        while (True):
+            self.wait_for_input()
+            if force_exit:
+                break
+            is_success, buffer = cv2.imencode(".jpg", self.input_frame)
+            request = predict_pb2.PredictRequest()
+            request.model_spec.name = 'text'
+            request.inputs['image'].CopyFrom(make_tensor_proto([buffer.tobytes()], shape=[1]))
+
+            predict_start_time = time.time()
+            result = stub.Predict(request, 10.0)
+            predict_duration = time.time() - predict_start_time
+            predict_duration *= 1000
+            self.predict_durations.append(predict_duration)
+
+            self.output_frame = draw_boxes(self.input_frame, result)
+            self.notify_output_ready()
+        print(f"Stopping requesting thread index: {self.index}")
+
+
+threads = [RequestingThread(i) for i in range(args.num_threads)]
+
+for thread in threads:
+    thread.start()
+
+
+def grab_frame(cap):
+    ret, frame = cap.read()
+    # crop square and resize if original image is too small for the model
+    if frame.shape[0] > 704 and frame.shape[1] > 704:
```

Comment:
I would state in readme for this client that we are effectively cropping the captured image to not have user wonder why is that happening.

Reply:
Or alternatively resize the original image and save proportions by cutting just the sides

Reply:
Added into readme